### PR TITLE
fix: push0 should increment program counter by 1 instead of 2

### PR DESCRIPTION
**Pull request by mandreyel**
## Description

The PR bumps the `sputnikvm` version, which contains a bugfix of the `shanghai` implementation.

## Testing

Added test which validates EVM transaction created with parameter: `evmVersion: 'shanghai'`.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -327,7 +327,7 @@ dependencies = [
 
 [[package]]
 name = "aurora-engine"
-version = "2.9.2"
+version = "2.10.0"
```
Comment (aleksuss) [inline Cargo.lock:330]: Why is this changed?

Comment (mandreyel) [reply Cargo.lock:330]: Not sure actually! I just ran `cargo update -p evm`.

Comment (mandreyel) [reply Cargo.lock:330]: Oh I know, it's because the version in `engine/Cargo.toml` has been changed to that version but apparently it hasn't been built since then and thus Cargo.lock was never updated. Think it's fine.

Comment (aleksuss) [reply Cargo.lock:330]: Yes. I see.

File: engine-tests/src/tests/sanity.rs [hunk 2]
```diff
@@ -694,6 +694,27 @@ fn test_eth_transfer_incorrect_nonce() {
     utils::validate_address_balance_and_nonce(&runner, dest_address, Wei::zero(), 0.into());
 }
 
+#[test]
+fn test_shanghai_contract() {
+    let (mut runner, mut source_account, _) = initialize_transfer();
+
+    // Panicked with `EvmFatal(Other("internal error: entered unreachable code [engine/src/engine.rs:148:54]"))`
```
Comment (birchmd) [inline engine-tests/src/tests/sanity.rs:701]: Probably don't need this comment anymore

File: engine-tests/src/tests/sanity.rs [hunk 3]
```diff
@@ -694,6 +694,27 @@ fn test_eth_transfer_incorrect_nonce() {
...
+                gas_price: 0.into(),
+                gas_limit: u64::MAX.into(),
+                to: None,
+                value: Wei::zero(),
+                data: hex::decode("6080604052348015600e575f80fd5b50607480601a5f395ff3fe6080604052348015600e575f80fd5b50600436106026575f3560e01c8063919840ad14602a575b5f80fd5b600560405190815260200160405180910390f3fea2646970667358221220cb01b9b9c75e5cd079a1980af2fe4397d2029888d12737d74cbbc10e0de65bd364736f6c63430008150033").unwrap(),
+            }
+        })
+        .unwrap();
+
+    println!("{result:?}");
```
Comment (birchmd) [inline engine-tests/src/tests/sanity.rs:715]: Println not needed, could think about adding an assert on the output instead or something.

File: engine-tests/src/tests/sanity.rs [hunk 4]
```diff
@@ -694,6 +694,27 @@ fn test_eth_transfer_incorrect_nonce() {
...
+            aurora_engine_transactions::legacy::TransactionLegacy {
+                nonce,
+                gas_price: 0.into(),
+                gas_limit: u64::MAX.into(),
+                to: None,
+                value: Wei::zero(),
+                data: hex::decode("6080604052348015600e575f80fd5b50607480601a5f395ff3fe6080604052348015600e575f80fd5b50600436106026575f3560e01c8063919840ad14602a575b5f80fd5b600560405190815260200160405180910390f3fea2646970667358221220cb01b9b9c75e5cd079a1980af2fe4397d2029888d12737d74cbbc10e0de65bd364736f6c63430008150033").unwrap(),
+            }
+        })
+        .unwrap();
```
Comment (birchmd) [inline engine-tests/src/tests/sanity.rs:713]:
```suggestion
        .expect("Should be able to execute EVM bytecode including PUSH0");
```

**Review by birchmd — APPROVED** [review]

**Review by aleksuss — APPROVED** [review]

**Review by mrLSD — APPROVED** [review]
